File: monster-battle.pl
#!/usr/bin/perl

use strict;
use warnings;

# Define the player's starting health and attack
my $player_health = 100;
my $player_attack = 20;

# Define the monster's starting health and attack
my $monster_health = 50;
my $monster_attack = 10;

# Main game loop
while ($player_health > 0 && $monster_health > 0) {
    # Display the player's health and the monster's health
    print "Player health: $player_health\n";
    print "Monster health: $monster_health\n";

    # Prompt the player to choose an action
    print "Choose an action (attack/run): ";
    my $action = lc(<STDIN>);
    chomp($action);

    # Handle the player's action
    if ($action eq "attack") {
        # Calculate the damage dealt by the player
        my $damage = int(rand($player_attack)) + 1;
        print "You deal $damage damage to the monster!\n";
        $monster_health -= $damage;

        # Check if the monster is dead
        if ($monster_health <= 0) {
            print "You have defeated the monster!\n";
            last;
        }

        # Calculate the damage dealt by the monster
        $damage = int(rand($monster_attack)) + 1;
        print "The monster deals $damage damage to you!\n";
        $player_health -= $damage;

        # Check if the player is dead
        if ($player_health <= 0) {
            print "You have been defeated by the monster.\n";
            last;
        }
    } elsif ($action eq "run") {
        # Calculate the chance of successfully escaping
        my $chance = int(rand(2));

        if ($chance == 0) {
            print "You failed to escape from the monster!\n";

            # Calculate the damage dealt by the monster
            my $damage = int(rand($monster_attack)) + 1;
            print "The monster deals $damage damage to you!\n";
            $player_health -= $damage;

            # Check if the player is dead
            if ($player_health <= 0) {
                print "You have been defeated by the monster.\n";
                last;
            }
        } else {
            print "You successfully escaped from the monster!\n";
            last;
        }
    } else {
        print "Invalid action. Please choose attack or run.\n";
    }
}

File: monster-battle.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

type monster struct {
	name   string
	health int
	attack int
}

func main() {
	rand.Seed(time.Now().UnixNano())

	playerMonster := monster{"Player Monster", 100, 20}
	enemyMonster := monster{"Enemy Monster", 100, 20}

	fmt.Println("A wild", enemyMonster.name, "appeared!")

	for playerMonster.health > 0 && enemyMonster.health > 0 {
		fmt.Println("\nPlayer Monster: ", playerMonster)
		fmt.Println("Enemy Monster: ", enemyMonster)

		fmt.Println("\nWhat will you do?")
		fmt.Println("1. Attack")
		fmt.Println("2. Heal")
		fmt.Print("Choose an action: ")
		var action int
		fmt.Scanln(&action)

		switch action {
		case 1:
			attack(&playerMonster, &enemyMonster)
		case 2:
			heal(&playerMonster)
		default:
			fmt.Println("Invalid action. Please try again.")
		}

		if enemyMonster.health > 0 {
			attack(&enemyMonster, &playerMonster)
		}
	}

	if playerMonster.health > 0 {
		fmt.Println("\nCongratulations! You defeated the enemy", enemyMonster.name)
	} else {
		fmt.Println("\nGame over. The enemy", enemyMonster.name, "defeated your monster.")
	}
}

func attack(attacker *monster, defender *monster) {
	damage := attacker.attack + rand.Intn(10)
	defender.health -= damage

	fmt.Println(attacker.name, "attacked", defender.name, "for", damage, "damage!")
}

func heal(monster *monster) {
	healAmount := 30 + rand.Intn(20)
	monster.health += healAmount

	if monster.health > 100 {
		monster.health = 100
	}

	fmt.Println(monster.name, "healed for", healAmount, "HP!")
}
